### ISSUE TITLE
confusionMatrix Feature Request: Marginal Map Proportion (weighting by base population)

### ISSUE THREAD

**Issue by mwip** [topepo/caret#1071]
Recently, I started adapting `caret::confusionMatrix` to fit state-of-the-art accuracy assessment for sampled land cover data to estimate marginal map area proportions as described in [Olofsson et al. (2014)](https://doi.org/10.1016/j.rse.2012.10.031) and [Stehman & Foody (2019)](https://doi.org/10.1016/j.rse.2019.05.018). 
Basically, the concept does rely on weighing the stratum in the error matrix by the proportion of each class in the the whole data set (in case you use stratified random sampling). 

This would involve adding a new parameter (`NULL` by default) holding the number of occurrences of each class in the parent population. If this vector is provided and fulfills the necessary properties, weighting by marginal proportion is conducted and the estimated confusion matrix as well as estimated overall and class accuracies are returned. 

Before I go all the way of filing a pull request, I wanted to know if there is any use for that outside of the remote sensing community. Perhaps this might be called somehow different, though. 
If you think this is of any value for the `caret` user base, I would be happy to create a PR. 

Let me know. I am looking forward to a fruitful discussion.

**Comment by topepo** [comment]
I just skimmed these papers. There could be use for it outside of land cover models (although I'm not sure that is true when you mix the area values in). It might be difficult to implement via `caret` though, unless a recipe is used. 

[Using a recipe](https://topepo.github.io/caret/using-recipes-with-train.html) could allow you to pass other variables into the performance assessment that are not used as outcomes or predictors. It also makes sure that the rows are matched during resampling. 

If you have code to do this/these metrics, it might make sense to have a PR into a package like `yardstick` or on the of the other performance metric packages. Then we could use it in `caret` in conjunction with a recipe.

**Comment by mwip** [comment]
The whole concept builds upon the assumption that the error matrix is calculated using only a sample of all occurrences. As land cover maps often have billions of pixels but there are only a few samples (let's say thousands). In my specific example, I had to assess the accuracy of such a map having a sample per every 40000 predicted pixels. The concept of marginal map proportions is used to estimate the accuracy of *all* pixels by weighting the confusion matrix of (thousands of) samples with the proportional occurrence of each class. 

Thus, the error matrix itself is weighted by the base population in the map. If I understand recipes correctly, this is not how they work. 

Since I already implemented it in a way inspired by the `caret::confusionMatrix` I'd create a PR to let you see what I mean.
